src/content/en/fundamentals/progressive-web-apps/app-shell.md

```diff
@@ -0,0 +1,268 @@
+not need to be loaded each time, but instead only gets the necessary content it
+needs from the network.
+
+For [single-page
+applications](https://en.wikipedia.org/wiki/Single-page_application) with
+JavaScript-heavy architectures, an application shell is a go-to approach. This
+approach relies on aggressively caching the shell (using [Service
+Worker](/web/fundamentals/primers/service-worker/)) to get the application
+running. Next, the dynamic content loads for each page using JavaScript. An app
+shell is useful for getting some initial HTML to the screen fast without a
+network.
+
+<img src="images/appshell.png" alt="Application Shell architecture" class="screenshot"/>
+
+Put another way, the app shell is similar to the bundle of code that you’d
+publish to an app store when building a native app. It is the skeleton of your
+UI and the core components necessary to get your app off the ground, but likely
+does not contain the data.
+
+Note: Try the [First Progressive Web
+App](https://codelabs.developers.google.com/codelabs/your-first-pwapp/#0)
+codelab to learn how to architectect and
+implement your first application shell for a weather app. The [Instant Loading with the App
+Shell model](https://www.youtube.com/watch?v=QhUzmR8eZAo) video also walks
+through this pattern.
+
+### When to use the app shell model
+
+Building a PWA does not mean starting from scratch. If you are building a modern
+single-page app, then you are probably using something similar to an app shell
+already whether you call it that or not. The details might vary a bit depending
+upon which libraries or frameworks you are using, but the concept itself is
+framework agnostic.
+
+An application shell architecture makes the most sense for apps and sites with
+relatively unchanging navigation but changing content. A number of modern
+JavaScript frameworks and libraries already encourage splitting your application
+logic from the content, making this architecture more straightforward to apply.
+For a certain class of websites that only have static content you can still
+follow the same model but the site is 100% app shell.
+
+To see how Google built an app shell architecture, take a look at
+[Building the Google I/O 2016 Progressive Web App](/web/showcase/2016/iowa2016).
+This real- world app started with a SPA to create a PWA that precaches content
+using a service worker, dynamically loads new pages, gracefully transitions
+between views, and reuses content after the first load.
+
+
+### Benefits {: #app-shell-benefits }
+
+The benefits of an app shell architecture with a service worker include:
+
+* **Reliable performance that is consistently fast**. Repeat visits are
+extremely quick.  Static assets (e.g. HTML, JavaScript, images and CSS) are
+immediately cached locally so there is no need to re-fetch the shell (and
+optionally the content if that is cached too). The UI is cached locally and
+content is updated dynamically as required.
```

Comment:
I forgot to change the default name of the commit. It's actually called 'Update app-shell.md.